src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -49,35 +53,44 @@ private MapHelper(List<T> source) {
          * @param flatMapping Функция преобразования элементов.
          */
         public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
-            // TODO реализация
-            throw new UnsupportedOperationException();
+            List<R> newSource = new ArrayList<>();
+            for (T t : source) {
+                for (R r : flatMapping.apply(t)) {
+                    newSource.add(r);
+                }
+            }
+            return new MapHelper<>(newSource);
         }
     }
 
     @Test
     public void mapEmployeesToLengthOfTheirFullNamesUsingMapHelper() {
         List<Employee> employees = Example1.getEmployees();
 
-        List<Integer> lengths = null;
-        // TODO                 MapHelper.from(employees)
-        // TODO                          .map(Employee -> Person)
-        // TODO                          .map(Person -> String(full name))
-        // TODO                          .map(String -> Integer(length of string))
-        // TODO                          .getMapped();
+        List<Integer> lengths = MapHelper.from(employees)
+                                   .map(Employee::getPerson)
+                                   .map(Person::getFullName)
+                                   .map(String::length)
+                                   .getMapped();
         assertEquals(Arrays.asList(14, 19, 14, 14), lengths);
     }
 
     @Test
     public void mapEmployeesToCodesOfLetterTheirPositionsUsingMapHelper() {
         List<Employee> employees = Example1.getEmployees();
 
-        List<Integer> codes = null;
-        // TODO               MapHelper.from(employees)
-        // TODO                        .flatMap(Employee -> JobHistoryEntry)
-        // TODO                        .map(JobHistoryEntry -> String(position))
-        // TODO                        .flatMap(String -> Character(letter))
-        // TODO                        .map(Character -> Integer(code letter)
-        // TODO                        .getMapped();
+        List<Integer> codes = MapHelper.from(employees)
+                                .flatMap(Employee::getJobHistory)
+                                .map(JobHistoryEntry::getPosition)
+                                .flatMap(string -> {
```

Comment:
Can be replaced with `s -> Arrays.asList(s.toCharArray())` or Guava's `Lists::charactersOf`

Reply:
I don't want to use Guava now and with s -> Arrays.asList(s.toCharArray()) it will return List<char[]>, not List<Character>, so it may be better to leave my implementation.